templates/fine-tune-llm_v2/cookbooks/long_context/README.ipynb

```diff
@@ -0,0 +1,115 @@
+    "TEST_DF = pd.read_csv(\"hf://datasets/Trelis/big_patent_60k_characters/test.csv\")\n",
+    "\n",
+    "# Fit this to how you want to construct your messages from your dataset. Pay attention to the names of columns from the dataset here.\n",
+    "def to_messages_dict(d: dict):\n",
+    "    \"\"\"Assembles a single example of the dataset for fine-tuning.\"\"\"\n",
+    "    return {\"messages\": [{\"role\": \"system\", \"content\": SYSTEM_MESSAGE}, {\"role\": \"user\", \"content\": d[\"description\"]},  {\"role\": \"assistant\", \"content\": d[\"abstract\"]}]}\n",
+    "\n",
+    "tokenizer = AutoTokenizer.from_pretrained(MODEL_NAME, token=HHUGGINFACE_TOKEN)\n",
+    "\n",
+    "def is_too_long(messages: dict):\n",
+    "    \"\"\"Filters out rows that exceed MAX_CONTEXT_LENGTH in their total length\"\"\"\n",
+    "    return sum([len(tokenizer(m[\"content\"])[\"input_ids\"]) for m in messages[\"messages\"]]) + SAFETY_BUFFER > MAX_CONTEXT_LENGTH\n",
+    "\n",
+    "for frame, output_file in [(TRAIN_DF, \"/mnt/cluster_storage/train.jsonl\"), (TEST_DF, \"/mnt/cluster_storage/test.jsonl\")]:\n",
+    "    with open(output_file, 'w') as f:\n",
+    "        for _, row in frame.iterrows():\n",
+    "            messages = to_messages_dict(row.to_dict())\n",
+    "            if not is_too_long(messages):\n",
+    "                json_str = json.dumps(messages)\n",
+    "                f.write(json_str + '\\n')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can now use this dataset to fine-tune an LLM that helps us with creating abstracts from patents.\n",
+    "The fine-tuned model will have a context-length of 8192 tokens during fine-tuning.\n",
+    "Check out the logs when fine-tuning to double-check every time you create a dataset this way!\n",
+    "\n",
+    "## FAQ:\n",
+    "\n",
+    "### How do I find out how many tokens the examples in my dataset have?\n",
+    "\n",
+    "If you fine-tune with this template, you will find exact statistics at the beginning of your fine-tuning job.\n",
+    "Exact numbers are hard to compute in advance. To get a rough idea, you can instantiate a tokenizer for your model and feed it a few samples from your dataset. Remember that one example consists of system message, a prompt, and an answer. There may also be [online services](https://belladoreai.github.io/llama3-tokenizer-js/example-demo/build/) that can help.\n",
+    "\n",
+    "### What if my dataset results in examples that are longer that the native context length of the model?\n",
+    "\n",
+    "Some datasets have examples that, when tokenize, exceed the context length of your LLM.\n",
+    "Anyscale Endpoints and the Anyscale platform support extending the native context length of LLMs. For example, for Llama 3 8B, we support fine-tuning with up to 32768 tokens (see [Anyscale Endpoints](https://docs.anyscale.com/canary/endpoints/fine-tuning/supported-models/) docs for a list).\n",
```

Comment:
Anyscale endpoints is gone. We should just completely remove any new references to that.